### ISSUE TITLE
Multiple email management

### ISSUE THREAD

**Issue by r-darwish** [magit/magit#3977]
This is more of a suggestion than a request. If accepted, I will open the PR. I use different emails with different Git repositories depending whether they are work related or personal.

In order to manage it from Emacs I wrote a [small function](https://github.com/drrlvn/.emacs.d/blob/1ee11a9e4745338a07d58be8127cf55d9f9c19bf/lisp/config-defuns.el#L447-L452) together with a custom variable. I think this will be a handy addition to Magit.

**Comment by tarsius** [comment]
> I think this will be a handy addition to Magit.

It certainly has its benefits that Magit provides many features that in other contexts would be provided by third-party packages. But it also has drawbacks not least that I end up having to maintain and improve lots of stuff I never use.

So yes, I am open to merging such a command but it has to be way beyond the minimal-viable state before I will do so. Completion for example should offer choices. That could also be useful for #3964. There's a function for setting Git variables; use that. The manual has to be updated. The release notes need an entry.

Also it would be nice if this were implemented as an transient infix. I.e. the value should be displayed in the committing popup. You can learn how to do that by looking at how it is done for other variables; the documentation probably won't be sufficient.

I don't mean to discourage you, but I would like you to think about all those things, preferably before you open a pull-request, because if you don't then I am going to have to do it. The minimal-viable version you have presented is useful, but clearly can be improved and I am certain if we added it like this, then sooner or later I would be asked to improve it.

**Comment by tarsius** [comment]
> This is more of a suggestion than a request.

But `feature-request` is never the less the most appropriate label we have.

**Comment by r-darwish** [comment]
This is not discouraging at all. This function was hacked to suite my needs, so it makes perfect sense that it has to be improved in order to become a part of Magit. 

If I understand correctly, here's my TODO list:
- [ ] Update the manual and the release notes.
- [ ] Implement this as a transient infix using `magit--git-variable:choices` to set the variable.

> Completion for example should offer choices. That could also be useful for #3964.

I'm not sure I understand. Currently the completion offers choices from `my/emails`. Should I offer choices from somewhere else other then a new `defcustom`?

**Comment by tarsius** [comment]
> Currently the completion offers choices from `my/emails`.

I overlooked that this is a list of addresses. A similar approach was suggested in the context of #3964. The difference is that here it is a list of "my identities" while there it is a list of "other people I sometimes work with". So if we in both cases did it as suggested, then we would end up with to variables that the user has to manually configure at some point.

I think it would be better if we used completion with (1) history and (2) completion candidates that were extracted from history. I would most likely prefer this approach, but it can't hurt to explore the approach with a list of fixed choices. While these choices would be fixed, they are not hard-coded and I don't remember whether `magit--git-variable:choices` can cope with that.
